logprep/metrics/metrics.py

```diff
@@ -156,43 +163,31 @@ def fullname(self):
     def init_tracker(self) -> None:
         """initializes the tracker and adds it to the trackers dict"""
         try:
-            if isinstance(self, CounterMetric):
-                self.tracker = Counter(
-                    name=self.fullname,
-                    documentation=self.description,
-                    labelnames=self.labels.keys(),
-                    registry=self._registry,
-                )
-            if isinstance(self, HistogramMetric):
-                self.tracker = Histogram(
-                    name=self.fullname,
-                    documentation=self.description,
-                    labelnames=self.labels.keys(),
-                    buckets=(0.00001, 0.00005, 0.0001, 0.001, 0.1, 1),
-                    registry=self._registry,
-                )
-            if isinstance(self, GaugeMetric):
-                self.tracker = Gauge(
-                    name=self.fullname,
-                    documentation=self.description,
-                    labelnames=self.labels.keys(),
-                    registry=self._registry,
-                    multiprocess_mode="liveall",
-                )
+            self._tracker = self._init_tracker()
         except ValueError as error:
             # pylint: disable=protected-access
-            self.tracker = self._registry._names_to_collectors.get(self.fullname)
+            tracker = self._registry._names_to_collectors.get(self.fullname)
             # pylint: enable=protected-access
-            if not isinstance(self.tracker, METRIC_TO_COLLECTOR_TYPE[type(self)]):
-                raise ValueError(
-                    f"Metric {self.fullname} already exists with different type"
-                ) from error
+            if tracker is not None:
+                if not isinstance(tracker, self.get_collector_type()):
```

Comment:
Following my suggestion below...

```suggestion
                if not isinstance(tracker, self.collector_type):
```